deepchem/models/torch_models/layers.py

```diff
@@ -4187,3 +4189,207 @@ def gaussian_histogram(self, x: torch.Tensor) -> torch.Tensor:
+    Let:
+        - **x_i** : ``The node to be updated``
+        - **h_i** : ``The hidden state of x_i``
+        - **x_j** : ``The neighbour node connected to x_i by edge e_ij``
+        - **h_j** : ``The hidden state of x_j``
+        - **W** : ``The edge weights``
+        - **m_ij** : ``The message between x_i and x_j``
+        - **h_j (self_loop)** : ``The set of hidden states of atom features``
+        - **mlp** : ``MultilayerPerceptron``
+        - **res** : ``ResidualBlock``
+
+    **In each message passing step**
+
+        .. code-block:: python
+            
+            m_ij = mlp1([h_i || h_j || e_ij])*(e_ij W)
+
+        **To handle self loops**
+
+            .. code-block:: python
+            
+                m_ij = m_ij + h_j(self_loop)
+
+    **In each update step**
+
+        .. code-block:: python
+
+            hm_j = res1(sum(m_ij)) 
+            h_j_new = mlp2(hm_j) + h_j 
+            h_j_new = res2(h_j_new) 
+            h_j_new = res3(h_j_new) 
+
+    .. note::
+    Message passing and message aggregation(sum) is handled by ``self.propagate()``.
+
+    References
+    ----------
+    .. [1] Molecular Mechanics-Driven Graph Neural Network with Multiplex Graph for Molecular Structures. https://arxiv.org/pdf/2011.07457.pdf
+
+
+    Examples
+    --------
+    The provided example demonstrates how to use the GlobalMessagePassing layer by creating an instance, passing input tensors (node_features, edge_attributes, edge_indices) through it, and checking the shape of the output.
+
+    Initializes variables and creates a configuration dictionary with specific values.
+
+    >>> dim = 1
+    >>> node_features = torch.tensor([[0.8343], [1.2713], [1.2713], [1.2713], [1.2713]])
+    >>> edge_attributes = torch.tensor([[1.0004], [1.0004], [1.0005], [1.0004], [1.0004],[-0.2644], [-0.2644], [-0.2644], [1.0004],[-0.2644], [-0.2644], [-0.2644], [1.0005],[-0.2644], [-0.2644], [-0.2644], [1.0004],[-0.2644], [-0.2644], [-0.2644]])
+    >>> edge_indices = torch.tensor([[0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4],[1, 2, 3, 4, 0, 2, 3, 4, 0, 1, 3, 4, 0, 1, 2, 4, 0, 1, 2, 3]])
+    >>> out = GlobalMessagePassing(dim)
+    >>> output = out(node_features, edge_attributes, edge_indices)
+    >>> output.shape
+    torch.Size([5, 1])
+
+    """
+
+    def __init__(self, dim: int):
+        """Initializes the GlobalMessagePassing layer.
+
+        Parameters
+        -----------
+        dim: int
+            The dimension of the input and output features.
+        """
+
+        super(GlobalMessagePassing, self).__init__()
+        self.dim = dim
+
+        self.h_mlp: MultilayerPerceptron = MultilayerPerceptron(
+            d_input=self.dim, d_output=self.dim, activation_fn='silu')
+
+        self.res1: ResidualBlock = ResidualBlock(self.dim)
+        self.res2: ResidualBlock = ResidualBlock(self.dim)
+        self.res3: ResidualBlock = ResidualBlock(self.dim)
+        self.mlp: MultilayerPerceptron = MultilayerPerceptron(
```

Comment:
Making the activation argument general instead of `silu` as advised by @ARY2260.

Reply:
Sure, I'll do.